app/main.py

```diff
@@ -1,2 +1,115 @@
+import math
+from dataclasses import dataclass
+from typing import Hashable, Any, Iterator
+
+
+@dataclass
+class Node:
```

Comment:
This class better be defined inside of a Dictionary since it is a part of its implementation and will not be used outside of it